osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineHitObjectBlueprint.cs

```diff
@@ -387,7 +387,7 @@ protected override void OnDrag(DragEvent e)
                         case IHasDuration endTimeHitObject:
                             var snappedTime = Math.Max(hitObject.StartTime, beatSnapProvider.SnapTime(time));
 
-                            if (endTimeHitObject.EndTime == snappedTime)
+                            if (endTimeHitObject.EndTime == snappedTime || (snappedTime - hitObject.StartTime) < 1)
```

Comment:
I am not sure why the added comparison is written this way. I assume it's because there's a worry of a precision issue, but I don't know whether it is actually an issue here, if `IBeatSnapProvider` quantises properly. In other words, if comparing directly was buggy, the preceding comparison would also be.

It would be good to have test coverage for this as well, if you can figure that out.

Reply:
I can work on getting a test class set up for this. For the comparison, I was noticing a precision issue when I was debugging. I was noticing that the if the `hitObject.StartTime` had a value of something like 16825 then `snappedTime` often times would be something like 16825.12756..., so I just added the subtraction comparison to ensure it still matches

Reply:
In that case, instead of explicitly doing that math, this should use `Precision.AlmostEquals` from framework (grep around for other usages). For both comparisons even, I'd say. Otherwise it looks as if these checks are half-buggy.

Reply:
Changed to use `Precision.AlmostEquals`. I'll start the test class work later today

Reply:
Resolving this conversation as the `AlmostEquals` has been added

Reply:
no need to add comments about it. would have eventually been resolved at some point anyway, most likely when I (or anyone else) is re-reviewing after tests have been added.